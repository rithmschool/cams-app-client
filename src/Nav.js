import React, {PropTypes, Component} from 'react'
import {Link} from 'react-router-dom'
import logo from './logo.png';
import './App.css';
import Dropdown, { DropdownTrigger, DropdownContent } from 'react-simple-dropdown';


class Nav extends Component {

  handleLogOut(e) {
    localStorage.removeItem('token');
  }

  static contextTypes = {
		router: PropTypes.object
	}

  edit(e) {
		let userId = JSON.parse(atob(localStorage.getItem('token').split('.')[1])).id
		this.context.router.history.push(`/users/${userId}/edit`)
	}

  render() {
    let login =
    <div>
      <button
        className="sign button dropdown-content button-hover">
        <Link to="/login">
          Login
        </Link>
      </button>
    </div>

    let logout =
    <div>
      <button className="sign button dropdown-content button-hover" onClick={this.edit.bind(this)}>
        Edit
      </button>
      <button
        className="sign button dropdown-content button-hover">
        <Link to="/logout">
          Log out
        </Link>
      </button>
    </div>

    let home =
    <Link to="/">
      <img className="logo button-hover" alt="Home" src={logo}/>
    </Link>

    let dashboard =
    <Link to="/dashboard">
      <img className="logo button-hover" alt="Dashboard" src={logo}/>
    </Link>

    return (
      <div className="nav">
        <div>
          {this.props.isLoggedIn ? dashboard : home}
        </div>
        <Dropdown className="account-dropdown" ref="dropdown">
          <DropdownTrigger>
            <button
              data-toggle="dropdown"
              className=" user button button-hover">
              <i className="fa fa-2x fa-user-circle-o"
                aria-hidden="true">
              </i>
            </button>
          </DropdownTrigger>
          <DropdownContent>
            {this.props.isLoggedIn ? logout : login}
          </DropdownContent>
        </Dropdown>
      </div>
    )
  }
}

export default Nav;
