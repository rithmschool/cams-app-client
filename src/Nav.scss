.dropdown__content {

  right: 0.5rem;

  button.button.dropdown-content {
    margin-top: 0;
  }

  .button-hover {
    display: block;
    padding: 0.7rem;
    width: 100%;
  }

}