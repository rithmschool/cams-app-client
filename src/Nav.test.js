import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import Nav from "./Nav";
import toJson from "enzyme-to-json";
import { Switch, MemoryRouter, Link } from "react-router-dom";

class LocalStorageMock {
  constructor() {
    this.store = { token: "nyc subway" };
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
}

global.localStorage = new LocalStorageMock();

test("nav bar snapshot when logged in", () => {
  const nav = renderer
    .create(
      <MemoryRouter initialEntries={["/"]} initialIndex={0}>
        <Nav isLoggedIn={true} />
      </MemoryRouter>
    )
    .toJSON();
  expect(toJson(nav)).toMatchSnapshot();
});

test("Nav bar has links when logged in", () => {
  const nav = mount(
    <MemoryRouter initialEntries={["/"]} initialIndex={0}>
      <Nav isLoggedIn={true} />
    </MemoryRouter>
  );
  console.log(nav.debug());
  expect(nav.find(Link)).toHaveLength(5);
  expect(nav.findWhere(node => node.props().to === "/")).toHaveLength(0);
  expect(nav.findWhere(node => node.props().to === "/dashboard")).toHaveLength(
    2
  );
  expect(
    nav.findWhere(node => node.props().to === "/playlists/new")
  ).toHaveLength(1);
  expect(
    nav.findWhere(node => node.props().to === "/assessments")
  ).toHaveLength(1);
  expect(nav.findWhere(node => node.props().to === "/login")).toHaveLength(1);
});

test("Nav bar has less links when logged out", () => {
  const nav = mount(
    <MemoryRouter initialEntries={["/"]} initialIndex={0}>
      <Nav isLoggedIn={false} />
    </MemoryRouter>
  );
  console.log(nav.debug());
  expect(nav.find(Link)).toHaveLength(2);
  expect(nav.findWhere(node => node.props().to === "/")).toHaveLength(1);
  expect(nav.findWhere(node => node.props().to === "/dashboard")).toHaveLength(
    0
  );
  expect(
    nav.findWhere(node => node.props().to === "/playlists/new")
  ).toHaveLength(0);
  expect(
    nav.findWhere(node => node.props().to === "/assessments")
  ).toHaveLength(0);
  expect(nav.findWhere(node => node.props().to === "/login")).toHaveLength(1);
});

test("nav bar snapshot when logged out", () => {
  const nav = renderer
    .create(
      <MemoryRouter initialEntries={["/"]} initialIndex={0}>
        <Nav isLoggedIn={false} />
      </MemoryRouter>
    )
    .toJSON();
  expect(toJson(nav)).toMatchSnapshot();
});
