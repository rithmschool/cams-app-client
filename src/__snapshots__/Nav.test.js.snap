exports[`test nav bar snapshot when logged in 1`] = `
<div
  className="nav">
  <div>
    <a
      href="/dashboard"
      onClick={[Function]}>
      <img
        alt="Dashboard"
        className="logo button-hover"
        src="logo.png" />
    </a>
  </div>
  <div>
    <div>
      <a
        href="/dashboard"
        onClick={[Function]}>
        <button
          className="button button-hover"
          type="submit"
          value="Submit">
          <i
            aria-hidden="true"
            className="fa fa-2x fa-home" />
        </button>
      </a>
      <a
        href="/playlists/new"
        onClick={[Function]}>
        <button
          className="button button-hover"
          type="submit"
          value="Submit">
          <i
            aria-hidden="true"
            className="fa fa-2x fa-plus" />
        </button>
      </a>
      <a
        href="/assessments"
        onClick={[Function]}>
        <button
          className="button button-hover"
          type="submit"
          value="Submit">
          <i
            aria-hidden="true"
            className="fa fa-2x fa-folder-open" />
        </button>
      </a>
    </div>
  </div>
  <div
    className="dropdown account-dropdown"
    style={undefined}>
    <a
      className="dropdown__trigger "
      onClick={[Function]}>
      <button
        className=" user button button-hover"
        data-toggle="dropdown">
        <i
          aria-hidden="true"
          className="fa fa-2x fa-user-circle-o" />
      </button>
    </a>
    <div
      className="dropdown__content ">
      <div>
        <button
          className="sign button dropdown-content button-hover"
          onClick={[Function]}>
          Edit
        </button>
        <button
          className="sign button dropdown-content button-hover"
          onClick={[Function]}>
          Invite Doctor
        </button>
        <button
          className="sign button dropdown-content button-hover">
          <a
            href="/login"
            onClick={[Function]}>
            Logout
          </a>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`test nav bar snapshot when logged out 1`] = `
<div
  className="nav">
  <div>
    <a
      href="/"
      onClick={[Function]}>
      <img
        alt="Home"
        className="logo button-hover"
        src="logo.png" />
    </a>
  </div>
  <div />
  <div
    className="dropdown account-dropdown"
    style={undefined}>
    <a
      className="dropdown__trigger "
      onClick={[Function]}>
      <button
        className=" user button button-hover"
        data-toggle="dropdown">
        <i
          aria-hidden="true"
          className="fa fa-2x fa-user-circle-o" />
      </button>
    </a>
    <div
      className="dropdown__content ">
      <div>
        <button
          className="sign button dropdown-content button-hover">
          <a
            href="/login"
            onClick={[Function]}>
            Login
          </a>
        </button>
      </div>
    </div>
  </div>
</div>
`;
